### PR TITLE
Feature/profile page

### PR DESCRIPTION
**Pull request by anetta999**
## Describe your changes

- created personal information page
- created personal information page navbar
- created edit personal information form
- created delete account modal

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://olga-team.atlassian.net/browse/TP-107)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit f

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

## Page
image
image
image
image

## Modal
image
image
image

### CODE REVIEW

File: src/components/cardInfoCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { Elements } from '@stripe/react-stripe-js';
+import { loadStripe } from '@stripe/stripe-js';
+import CheckoutForm from './checkoutForm';
+import stripe from 'stripe';
+
+const stripePromise = loadStripe(
+  import.meta.env.VITE_STRIPE_PUBLIC_KEY as string,
+);
+console.log(import.meta.env.VITE_STRIPE_PUBLIC_KEY);
```
Comment (eugeneiv1) [inline src/components/cardInfoCard/index.tsx]: ?

File: src/components/deleteAccountModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { MouseEvent, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import { DeleteAccountModalProps } from './types';
+import { useTranslation } from 'react-i18next';
+import { Backdrop, Modal, CancelBtn, DeleteBtn, CloseBtn } from './styles';
+import DeleteIcon from 'assets/icons/iconDelete';
+import CloseIcon from 'assets/icons/iconClose';
+
```
Comment (eugeneiv1) [inline src/components/deleteAccountModal/index.tsx:9]: is import order correct?

File: src/components/deleteAccountModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const modalRoot = document.querySelector('#modal-root');
+
+export const DeleteAccountModal = ({
+  isModalOpen,
+  onClose,
+}: DeleteAccountModalProps) => {
+  const { t } = useTranslation();
+  useEffect(() => {
```
Comment (eugeneiv1) [inline src/components/deleteAccountModal/index.tsx:22]: para

File: src/components/deleteAccountModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    const handleModalCloseByEsc = (evt: KeyboardEvent) => {
+      if (evt.code === 'Escape') {
+        onClose();
```
Comment (eugeneiv1) [inline src/components/deleteAccountModal/index.tsx:25]: compare to string

File: src/components/deleteAccountModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  return createPortal(
+    <>
+      <Backdrop onClick={handleModalCloseByClickOnBackdrop}>
+        <Modal>
+          <CloseBtn type="button" onClick={onClose}>
+            <CloseIcon />
+          </CloseBtn>
+          <DeleteIcon width={94} height={94} />
+          <h1>{t('deleteAccount.title')}</h1>
```
Comment (eugeneiv1) [inline src/components/deleteAccountModal/index.tsx:60]: mb size of icon should be moved to const?

File: src/components/editProfileForm/editProfileFormHooks.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import * as yup from 'yup';
+import { useTranslation } from 'react-i18next';
+const maxCharactersLength: number = 100;
```
Comment (eugeneiv1) [inline src/components/editProfileForm/editProfileFormHooks.ts:4]: para before

File: src/components/editProfileForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+} from './data/sizes';
+import { SubmitHandler, useForm, Controller } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+import { Inputs } from './types';
+import PhoneInput from 'react-phone-number-input';
+import 'react-phone-number-input/style.css';
+import Select from 'react-select';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEditProfileSchema } from './editProfileFormHooks';
+import CreditCardForm from 'components/cardInfoCard';
```
Comment (eugeneiv1) [inline src/components/editProfileForm/index.tsx]: import order

File: src/components/editProfileForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+
+  return (
+    <>
+      <EditForm onSubmit={handleSubmit(onSubmit)}>
+        <ImgWrap>
+          <img
+            src={avatar ? avatar : defaultAvatar}
+            alt="user-avatar"
+            width={120}
+            height={120}
```
Comment (eugeneiv1) [inline src/components/editProfileForm/index.tsx]: inline styles?

File: src/components/editProfileForm/index.tsx [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+              <label htmlFor="user-phone">{t('editProfile.phone')}</label>
+              <Controller
+                name="phone"
+                control={control}
+                render={({ field }) => (
+                  <PhoneInput
+                    {...field}
+                    international
+                    defaultCountry="US"
+                    countries={['US']}
```
Comment (eugeneiv1) [inline src/components/editProfileForm/index.tsx]: mb move to const?

File: src/components/profileNavBar/index.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+import { ProfileNavBarProps } from './types';
+
+export const ProfileNavBar = ({ toggleModal }: ProfileNavBarProps) => {
+  const { t } = useTranslation();
+  const [isVendor, setIsVendor] = useState<boolean>(false);
+
+  const handleUserMode = () => {
+    setIsVendor(!isVendor);
+  };
+  return (
```
Comment (eugeneiv1) [inline src/components/profileNavBar/index.tsx:34]: para

File: src/components/profileNavBar/index.tsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        )}
+        <li>
+          <EvenNavLink to="">
+            <SupportIcon width={32} height={32} />
+            <span>{t('profileNav.navLinks.support')}</span>
+          </EvenNavLink>
+        </li>
+        <li>
+          <OddNavLink to="">
+            <LogoutIcon width={32} height={32} />
```
Comment (eugeneiv1) [inline src/components/profileNavBar/index.tsx]: mb sizes move to const?

File: src/const/paths.ts [hunk 12]
```diff
@@ -11,5 +11,7 @@ export const paths = {
   verifyOtp: '/verify-otp',
   verifyEmail: '/verify-email',
   getUser: '/users',
+  profile: '/profile',
+  personalInfo: '/profile',
```
Comment (eugeneiv1) [inline src/const/paths.ts:16]: 2 identical paths??

File: src/pages/profilePage/index.tsx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+import {
+  OutletWrap,
+  ProfileNavBarWrap,
+  ProfileSection,
+  ProfileSectionWrap,
+  TitleSection,
+} from './styles';
+import { useTranslation } from 'react-i18next';
+import { useState } from 'react';
+import { DeleteAccountModal } from 'components/deleteAccountModal';
```
Comment (eugeneiv1) [inline src/pages/profilePage/index.tsx]: import order

File: .env [hunk 14]
```diff
@@ -0,0 +1,2 @@
+VITE_STRIPE_PUBLIC_KEY='pk_test_51Nhf4NLtxhLqEU8X8yobOruIFWfeBEwp5xQF7OEuVx2pSGMUWzfQ8LLzPYU4UZoOiDCv0v096VLwffvMLLGQL0hQ005wLyXDmR'
```
Comment (AlexandraVoloshyna) [inline .env]: env should be in gitignore

Comment (anetta999) [reply .env]: fixed in [fix: added env to gitignore file and stop tracking it](https://github.com/ZenBit-Tech/faq_team_fe/pull/11/commits/46a2243175a98b53a959f68792e11bc9fd85f26f)

File: src/assets/icons/editIcon.tsx [hunk 15]
```diff
@@ -0,0 +1,31 @@
+import { SVGProps } from 'react';
+const EditIcon = (props: SVGProps<SVGSVGElement>) => (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" {...props}>
+    <g clipPath="url(#clip0_822_63665)">
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M17.7375 3.89831C16.7909 2.94405 15.269 2.94405 14.3224 3.89831L5.1253 13.1454C4.90184 13.3768 4.77062 13.6499 4.73337 13.9292L4.73259 13.9351L4.22833 17.5015L7.12997 17.0892V17.0735L7.78084 16.9867C8.06311 16.9491 8.32423 16.8182 8.52382 16.6061L8.53091 16.5986L17.7382 7.34123L17.7397 7.33979C18.6856 6.39385 18.6886 4.85709 17.7375 3.89831ZM13.258 2.8414C14.7914 1.29618 17.2692 1.29636 18.8024 2.84192C20.3311 4.38288 20.3344 6.86555 18.8011 8.39968C18.8008 8.39993 18.8006 8.40019 18.8003 8.40045L9.60846 17.6422C9.32902 17.9368 8.99446 18.1597 8.62997 18.3041V18.3911L4.1155 19.0327L4.11315 19.033C3.73627 19.0853 3.33714 18.9856 3.04412 18.6962C2.74942 18.405 2.64482 18.0038 2.69672 17.6197L2.69734 17.6151L3.24696 13.7278C3.3301 13.1096 3.61775 12.545 4.0518 12.0978L4.05818 12.0912L13.258 2.8414C13.2582 2.84123 13.2578 2.84157 13.258 2.8414Z"
+        // fill="#6D6B6B"
```
Comment (MishaFomenko) [inline src/assets/icons/editIcon.tsx]: commented code

File: src/components/deleteAccountModal/index.tsx [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+const deleteIconSize: number = 94;
+
+export const DeleteAccountModal = ({
+  isModalOpen,
+  onClose,
+}: DeleteAccountModalProps) => {
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    const handleModalCloseByEsc = (evt: KeyboardEvent) => {
```
Comment (MishaFomenko) [inline src/components/deleteAccountModal/index.tsx]: output type

File: src/components/deleteAccountModal/index.tsx [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+      document.body.style.overflow = hiddenOverflow;
+    }
+
+    return () => {
+      document.body.style.overflow = autoOverflow;
+      window.removeEventListener('keydown', handleModalCloseByEsc);
+    };
+  }, [onClose, isModalOpen]);
+
+  const handleModalCloseByClickOnBackdrop = (evt: MouseEvent<HTMLElement>) => {
```
Comment (MishaFomenko) [inline src/components/deleteAccountModal/index.tsx]: output type

File: src/components/editProfileForm/index.tsx [hunk 18]
```diff
@@ -0,0 +1,325 @@
...
+import 'react-phone-number-input/style.css';
+
+import {
+  cities,
+  clothesSizes,
+  jeansSizes,
+  shoesSizes,
+  states,
+} from './data/sizes';
+import { useEditProfileSchema } from './editProfileFormHooks';
```
Comment (MishaFomenko) [inline src/components/editProfileForm/index.tsx:22]: absolute imports

File: src/components/footer/index.tsx [hunk 19]
```diff
@@ -1,25 +1,26 @@
...
+  FooterMenuList,
+  FooterRight,
   FooterSocials,
   FooterText,
   FooterTitle,
-  FooterMenuList,
-  FooterIcon,
-  Credentials,
+  StyledFooter,
 } from './styles';
```
Comment (MishaFomenko) [inline src/components/footer/index.tsx:23]: absolute import

File: src/components/mainHeading/index.tsx [hunk 20]
```diff
@@ -1,4 +1,4 @@
-import { StyledH1, HeadingInner } from './styles';
+import { HeadingInner, StyledH1 } from './styles';
```
Comment (MishaFomenko) [inline src/components/mainHeading/index.tsx:1]: absolute imports

File: src/components/profileNavBar/index.tsx [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+      <NavList>
+        <li>
+          <OddNavLink to={paths.personalInfo}>
+            <UserIcon width={iconsSize} height={iconsSize} />
+            <span>{t('profileNav.navLinks.info')}</span>
+          </OddNavLink>
+        </li>
+        {isVendor ? null : (
+          <li>
+            <EvenNavLink to="">
```
Comment (MishaFomenko) [inline src/components/profileNavBar/index.tsx]: links should be extracted into a separate file

File: src/components/profileNavBar/styles.ts [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+  }
+`;
+
+export const StyledNavLink = styled(NavLink)<{ theme?: Theme }>`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  border-radius: 8px;
+  font-weight: ${({ theme }) => theme.fontWeight.regular};
+  font-size: 16px;
```
Comment (MishaFomenko) [inline src/components/profileNavBar/styles.ts]: constant

**Pull request merged**